src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationPerformanceMonitor.cs

```diff
@@ -28,22 +48,48 @@ public struct ParsedMessage
         {
             public bool PerformanceMonitoringEnabled;
             public List<Tuple<string, double>> EventDurations;
+            public List<Tuple<string, double>> SummedEventDurations;
             public List<Tuple<string, int>> EventCounts;
+            public List<Tuple<string, MemoryUsage>> MemoryUsages;
 
-            public ParsedMessage(bool performanceMonitoringEnabled, List<Tuple<string, double>> eventDurations, List<Tuple<string, int>> eventCounts)
+            public ParsedMessage(bool performanceMonitoringEnabled, List<Tuple<string, double>> eventDurations, List<Tuple<string, double>> summedDurations, List<Tuple<string, int>> eventCounts, List<Tuple<string, MemoryUsage>> memoryUsages)
             {
                 this.PerformanceMonitoringEnabled = performanceMonitoringEnabled;
                 this.EventDurations = eventDurations;
+                this.SummedEventDurations = summedDurations;
                 this.EventCounts = eventCounts;
+                this.MemoryUsages = memoryUsages; 
             }
+
+            public static ParsedMessage Empty => new ParsedMessage(false, null, null, null, null);
         };
 
-        private Dictionary<StopWatchKey, Stopwatch> eventStopWatches = new Dictionary<StopWatchKey, Stopwatch>();
-        private Dictionary<StopWatchKey, int> eventCounts = new Dictionary<StopWatchKey, int>();
+        private Dictionary<PerformanceEventKey, Stopwatch> eventStopWatches = new Dictionary<PerformanceEventKey, Stopwatch>();
+        private Dictionary<PerformanceEventKey, Stopwatch> incrementEventStopWatches = new Dictionary<PerformanceEventKey, Stopwatch>();
+        private Dictionary<PerformanceEventKey, int> eventCounts = new Dictionary<PerformanceEventKey, int>();
+        private Dictionary<PerformanceEventKey, MemoryUsage> eventMemoryUsage = new Dictionary<PerformanceEventKey, MemoryUsage>();
 
         protected override void Awake()
         {
             base.Awake();
+            Profiler.enabled = StateSynchronizationPerformanceParameters.EnablePerformanceReporting;
+        }
+
+        public IDisposable IncrementEventDuration(string componentName, string eventName)
+        {
+            if (!StateSynchronizationPerformanceParameters.EnablePerformanceReporting)
+            {
+                return null;
+            }
+
+            var key = new PerformanceEventKey(componentName, eventName);
+            if (!incrementEventStopWatches.TryGetValue(key, out var stopwatch))
+            {
+                stopwatch = new Stopwatch();

```

Comment:
Can we avoid all these stopwatch objects? And therefore the map lookup also. Have a single static stopwatch and record the start/end ticks?

Reply:
This seems like a good suggestion, i will look into this now.

Reply:
At this point in time, we could ditch the stopwatch allocations and use one static stopwatch, but there would like be some sort of class object that is handed to the TimeScope and MemoryScope every event call. I've opened the following issue for this:
https://github.com/microsoft/MixedReality-SpectatorView/issues/288